pkg/chains/formats/intotoite6/intotoite6.go

```diff
@@ -151,6 +151,21 @@ func GetSubjectDigests(tr *v1beta1.TaskRun, logger *zap.SugaredLogger) []intoto.
 		}
 	}
 
+	sts := artifacts.ExtractSignableTargetFromResults(tr, logger)
+	for _, obj := range sts {
+		splits := strings.Split(obj.Digest, ":")
+		if len(splits) != 2 {
+			logger.Errorf("Digest %s should be in the format of: algorthm:abc", obj.Digest)
+			continue
+		}
+		subjects = append(subjects, intoto.Subject{
```

Comment:
Do we confirm that these artifacts exist before adding them as subjects anywhere? I'm asking because I'm wondering if you could configure a TaskRun's results to say something was created when it wasn't, and then add it as a Subject to the attestation.

Right now we only support OCI images and if the image doesn't exist then we can't store the attestation alongside it, so it kind of gets around that.

Reply:
Even if we did verify existence, a malicious task could lie and point to an existing version. 🤔 

We'd probably need to couple this with Spire + Trusted Resources in order to trust the value.

Reply:
SPIRE would guarantee the result hasn't been modified, but not that the artifact actually exists. I don't know much about Trusted Resources, would adding that in provide that guarantee? If not, I think Chains is the only spot we can make sure the artifact exists (agree it's not perfect, but it does come with some nice guarantees). 

At the moment, the Subjects themselves are also signed before the signed attestation is generated and attached because they are an artifact Chains recognizes (`oci`). The new Subjects added here wouldn't be signed, and we wouldn't know if they exist or not. I wonder if something like what I described in https://github.com/tektoncd/chains/issues/276#issuecomment-959295196 might be the move. Then for an arbitrary file we could sign it, create the attestation and store that somewhere.